components/cap-reporter.js

```diff
@@ -0,0 +1,49 @@
+import { LitElement, html, css } from "../lib/lit.js";
+import { fetchVolumesData, fetchReporterData } from "../lib/data.js";
+
+export default class CapReporter extends LitElement {
+	static properties = {
+		volumesData: { attribute: false },
+		reporterData: { attribute: false },
+		reporter: { type: String },
+	};
+
+	constructor() {
+		super();
+		this.volumesData = [];
+		this.reporterData = {};
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		fetchVolumesData(this.reporter, this);
```

Comment:
> @mdellabitta: A point I'm not super pleased about is that each of those data fetching functions passes the component in as a function parameter and sets a parameter on the component when the fetch promise resolves. This doesn't seem super idiomatic and has the unfortunate side effect of coupling the fetching function to the component in an unneccesary way. 

I think this is very okay. I that bothers you, you could alternatively use inheritance to handle this. 

Provided Lit lets you inherit properties: 

```javascript
export default class CapDataRenderer extends LitElement { 
  static properties = {
    casesData: { attribute: false },
    volumesData: { attribute: false },
    reporterData: { attribute: false },
  }
  
  ... 
  
  fetchCasesList() { ... }
  fetchVolumesData() { ... }
  fetchReporterData() { ... }
 }
```

You could then have `CapReporter`, `CapJuridictions`, and `CapVolume` inherit these methods and properties.  I would even consider making the attributes generic in that context.

Reply:
I agree with these sentiments — I'm not opposed to what you have, but it was a little less intuitive to me to need to dig into the methods themselves to see how data was being set on component, rather than see a component set its own internal data based on what is returned by a method it is called or by the data that it is passed.